students/Sally_Shen/lesson3/mailroom.py

```diff
@@ -0,0 +1,96 @@
+
+
+'''choms'''
+
+import sys
+
+print("in mailroom")
+'''Writing first name is easier for testing.'''
+donors ={"Fred Gates": [100.90, 200, 50], "Jeff Bazos": [100, 50, 600], "Mark Zuckerberg": [200, 50, 500],
+         "Paul Allen": [300, 40, 200]}
+
+
+def thank_you():
+    response = input("Type your donor name, if don't know, type list to pick")
+    '''If the user types list show them a list of the donor names and re-prompt'''
+    if response == "list":
+        for key in donors.keys():
+            print(key)
+        response = input("Please type your donor name.")
+
+    if response not in donors.keys():
+        donors[response] = []
+
+    newDonation = input("how much to donate?")
+    donors[response].append(float(newDonation))
+    email = '''
+    Dear {0},
+    Thank you for donating {1}!\n'''.format(response, newDonation)
+    print(email)
+
+
+def gen_stats(donor):
+    donations = donor[1]
+    total = sum(donations)
+    num  =  len(donations)
+    stats = (donor[0], total, num, total/num)
+    return
+
+def report():
+    amountToNames = {}
+    for name, amount in donors.items():
+        total = sum(amount)
+        if total in amountToNames.keys():
```

Comment:
you are building this from scratch -- it won't alrady be there. And if it is, it's coincidence -- what if two donor happen to have the same total?

Ahh -- I see what you are doing here -- you are building a lit of names with the same total, so that you can capture duplicates. That can work, yes, but it's getting a bit too complicated.

Really better to create a list of al the donor information, and custom sort that.